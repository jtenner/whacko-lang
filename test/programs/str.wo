export fn _start(): void {
  let a = "Hello world";
}