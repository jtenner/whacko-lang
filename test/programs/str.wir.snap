program:
  callable:[test/programs/str.wo]_start;functionType:()=>voidType
    block: entry~35
      36~inst = new: classType:[std/str.wo]str<>
      38~inst = (str "Hello world")
      39~inst = ptrtoint: (runtime classType:[std/str.wo]str<> 38~inst)
      40~inst = (u32 11)
      41~inst = call: callable:[std/str.wo]str.constructor;methodType:classType:[std/str.wo]str<>#(usizeType,usizeType)=>voidType with (runtime classType:[std/str.wo]str<> 36~inst), (runtime usizeType 39~inst), (runtime usizeType 40~inst)
      42~inst = store: (variable a) = (runtime classType:[std/str.wo]str<> 36~inst)
      return (void)

  callable:[std/str.wo]str.constructor;methodType:classType:[std/str.wo]str<>#(usizeType,usizeType)=>voidType
    block: entry~44
      45~inst = load: (field classType:[std/str.wo]str<>#ptr)
      46~inst = load: (variable ptr)
      47~inst = store: (field classType:[std/str.wo]str<>#ptr) = (runtime usizeType 46~inst)
      48~inst = load: (field classType:[std/str.wo]str<>#size)
      49~inst = load: (variable size)
      50~inst = store: (field classType:[std/str.wo]str<>#size) = (runtime usizeType 49~inst)
      return (void)

